src/libstd/sys/unix/thread.rs

```diff
@@ -126,6 +125,20 @@ impl Thread {
         }
     }
 
+    #[cfg(target_os = "netbsd")]
+    pub fn set_name(name: &str) {
+        extern {
+            fn pthread_setname_np(thread: libc::pthread_t,
+                                  name: *const libc::c_char,
+                                  arg: *mut libc::c_void) -> libc::c_int;
+        }
+        let cname = CString::new(&b"%s"[..]).unwrap();
+        let carg = CString::new(name).unwrap().as_ptr() as *mut libc::c_void;
```

Comment:
This'll end up being a use-after-free because the `CString` needs to live over the call of `pthread_setname_np`